### PR TITLE
feat(tests): validate state and nonce in ethereum vmtests

### PR DESCRIPTION
**Pull request by magicalne**
The Ethereum tests validate three parts:
- state root hash
- nonce
- logs hash

The logs hash validation was implemented before. And comparing nonce of accounts is straightforward.

The state validation is tricky because we use a different Merkle tree implementation.
Luckily, we can compare storage key-value just from filler files.

### CODE REVIEW

**Comment by Flouse** [comment]
# Review notes

## skipped tests
```rs
// Explain why we skip those tests.
const EXCLUDE_VMTEST_FILES: &[&str] = &[
    "sha3.json",      // Failure: memory size issue.
    "msize.json",     // Failure: memory size issue.
    "gas.json",       // Failure: memory size issue.
    "blockInfo.json", // Failure: Some fields are mocked in polyjuice. It's not testable.
    "loopMul.json",   // Success but too slow.
    "loopExp.json",   // Success but too slow.
];
```

File: .github/workflows/gwos-evm-ci.yml [hunk 1]
```diff
@@ -52,7 +52,7 @@ jobs:
 
     - name: Run Ethereum Test
       working-directory: gwos-evm/polyjuice-tests
-      run: RUST_LOG=info,gw_generator=debug cargo test --release --test ethereum_test -- ethereum_test --nocapture
+      run: RUST_LOG=info,gw_generator=debug cargo test --release --test ethereum_test -- ethereum_vm_test --nocapture
```
Comment (Flouse) [inline .github/workflows/gwos-evm-ci.yml:55]:
```
test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 3 filtered out; finished in 0.00s
```

> https://github.com/godwokenrises/godwoken/actions/runs/3928517927/jobs/6716226374

**Review by Flouse — CHANGES_REQUESTED** [review]

File: gwos-evm/polyjuice-tests/tests/ethereum_test.rs [hunk 2]
```diff
@@ -287,83 +447,134 @@ fn hex_to_eth_address(hex_str: &str) -> anyhow::Result<[u8; 20]> {
...
-                    println!("Starting test with: {:?}", &path);
-                    let content = fs::read_to_string(&path)?;
-                    let test_cases: HashMap<String, TestCase> = serde_json::from_str(&content)?;
-                    for (testname, testcase) in test_cases {
-                        println!("test name: {}", testname);
-                        let runner = VMTestRunner::new(testcase)?;
-                        if runner.run().is_err() {
-                            err_cases.push(path.clone());
-                        }
+fn ethereum_vmtest_test() -> anyhow::Result<()> {
```
Comment (Flouse) [inline gwos-evm/polyjuice-tests/tests/ethereum_test.rs:512]:
```suggestion
fn ethereum_vm_test() -> anyhow::Result<()> {
```

**Review by Flouse — CHANGES_REQUESTED** [review]

File: .github/workflows/gwos-evm-ci.yml [hunk 3]
```diff
@@ -52,7 +52,7 @@ jobs:
 
     - name: Run Ethereum Test
       working-directory: gwos-evm/polyjuice-tests
-      run: RUST_LOG=info,gw_generator=debug cargo test --release --test ethereum_test -- ethereum_test --nocapture
```
Comment (Flouse) [inline .github/workflows/gwos-evm-ci.yml:55]: Why skip [`ethereum_test`](https://github.com/godwokenrises/godwoken/blob/31c4f568ebf92412ecc8ca60c2bef36a997022ac/gwos-evm/polyjuice-tests/tests/ethereum_test.rs#L472-L509)?

Comment (magicalne) [reply .github/workflows/gwos-evm-ci.yml:55]: Here is a simple rename. We never run the whole Ethereum test. Just the VMTests directory.
BTW, the entire Ethereum tests are still in testing.

Comment (Flouse) [reply .github/workflows/gwos-evm-ci.yml:55]: I see

**Review by Flouse — DISMISSED** [review]

**Review by Flouse — DISMISSED** [review]

**Comment by jjyr** [comment]
Please resolve conflicts

**Comment by magicalne** [comment]
> Please resolve conflicts

Fixed.

**Review by Flouse — APPROVED** [review]

**Pull request merged**